pkg/validation/internal/community.go

```diff
@@ -0,0 +1,277 @@
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	// Check if has deprecated apis then, check the olm.maxOpenShiftVersion property
+	if len(bundle.V1beta1CRDs) > 0 {
+		checks := CSVChecks{csv: *bundle.CSV, errs: []error{}, warns: []error{}}
+		checks = checkMaxOpenShiftVersion(checks)
+		checks = checkOCPLabelsWithHasDeprecatedAPIs(checks, indexImagePath)
+		for _, err := range checks.errs {
+			result.Add(errors.ErrInvalidCSV(err.Error(), bundle.CSV.GetName()))
+		}
+		for _, warn := range checks.warns {
+			result.Add(errors.WarnInvalidCSV(warn.Error(), bundle.CSV.GetName()))
+		}
+	}
+
+	return result
+}
+
+type propertiesAnnotation struct {
+	Type  string
+	Value string
+}
+
+// checkMaxOpenShiftVersion will verify if the OpenShiftVersion property was informed
+func checkMaxOpenShiftVersion(checks CSVChecks) CSVChecks {
+	// Ensure that has the OCPMaxAnnotation
+	const olmproperties = "olm.properties"
+	const olmmaxOpenShiftVersion = "olm.maxOpenShiftVersion"
+	semVerOCPV1beta1Unsupported, _ := semver.ParseTolerant(ocpVerV1beta1Unsupported)
+
+	properties := checks.csv.Annotations[olmproperties]
+	if len(properties) == 0 {
+		checks.errs = append(checks.errs, fmt.Errorf("csv.Annotations not specified %s with for an "+
+			"OCP version < %s. This annotation is mandatory to prevent the user from upgrading their OCP cluster "+
+			"before they have installed a version of your operator which is compatible with %s",
+			olmmaxOpenShiftVersion, ocpVerV1beta1Unsupported, ocpVerV1beta1Unsupported))
+		return checks
+	}
+
+	var properList []propertiesAnnotation
+	if err := json.Unmarshal([]byte(properties), &properList); err != nil {
+		checks.errs = append(checks.errs, fmt.Errorf("csv.Annotations has an invalid value specified for %s. "+
+			"Please, check the value informed (%s) and ensure that is an array such as: "+
+			"\"olm.properties\": '[{\"type\": \"key name\", \"value\": \"key value\"}]'",
+			olmproperties, properties))
+		return checks
+	}
+
+	hasOlmMaxOpenShiftVersion := false
+	olmMaxOpenShiftVersionValue := ""
+	for _, v := range properList {
+		if v.Type == olmmaxOpenShiftVersion {
+			hasOlmMaxOpenShiftVersion = true
+			olmMaxOpenShiftVersionValue = v.Value
+			break
+		}
+	}
+
+	if !hasOlmMaxOpenShiftVersion {
+		checks.errs = append(checks.errs, fmt.Errorf("csv.Annotations.%s with the "+
+			"key and value %s and an OCP version which is < %s is mandatory for any operator bundle which is using the "+
+			"deprecated apis for CRDs (apiextensions.k8s.io/v1beta1) ",
+			olmproperties, olmmaxOpenShiftVersion, ocpVerV1beta1Unsupported))
+		return checks
+	}
+
+	semVerVersionMaxOcp, err := semver.ParseTolerant(olmMaxOpenShiftVersionValue)
+	if err != nil {
+		checks.errs = append(checks.errs, fmt.Errorf("csv.Annotations.%s with invalid value informed for "+
```

Comment:
I don't know what you mean by "invalid value informed for". The word "informed" seems weird in this context but I don't know what to suggest as I'm not quite sure what you are trying to convey.

Reply:
it is in the case of the user use the annotation and then, inform a value which is not e.g. 4.8. I changed that. Tks.